CIP-dapp-bridge-rework/README.md

```diff
@@ -0,0 +1,1169 @@
+
+By solving these problems, we will lower the entry barrier for DApp developers into the ecosystem, while simultaneously enhancing the security of the end-user who has entrusted their funds to a wallet, but should not trust their management to just any application they encounter. The wallet will act as a black box, and the DApp as an extension of its UI, not a replacement of its engine.
+
+Additionally, this approach will simplify the integration of wallets and DApps from the wallet developers' perspective, as it will provide a more natural interface for interaction, easily translatable into native scenarios of interaction with the ordinary user through the wallet's own UI.
+
+## Specification
+
+### The main idea
+The main idea is to delegate as many operations as possible to the wallet, which are its core functions. This will greatly simplify the lives of DApps developers. It will also normalize the transaction pipeline, which is currently highly intermingled between the DApp and the wallet.
+We are moving away from imperative interaction with the wallet towards a declarative approach, where the DApp does not have to assemble the transaction itself; it should be sufficient to **describe** the desired outcome. Just as a regular wallet user is not required to delve into all the details - they simply specify the address, amount, and press a button. This is literally why software wallets exist.
+For example, here is a description of a transaction sending 1 ada to some address in the testnet:
+
+```
+{
+    outputs:[{
+        address: "addr_test1qpy486p7jhq933kay7rf78jz2j6k5ya7u95nnnjetgj9r02dp276p7j8023vmum9wu8gp7q54f3rjke45j0klk3pmwvsgyx4n7",
+        value: {
+            lovelaces: 1000000
+        }
+    }]
+}
+```
+
+Or a description of a delegation transaction to a certain pool:
+
+
+```
+{
+    certificates: [{
+        type: "stakeDelegation",
+        stakeKeyId: 0,
+        poolHash: "660f2b816e90e4e826daebf33357d76ab89414f88d57dd7d9f7898fe"
+    }]
+}
+```
+
+
+This is quite enough. The DApp shouldn't have to calculate change, fees, select inputs, withdrawal rewards and so on. It just needs to **tell the wallet what result it needs**.
+Let's understand how this works.
+
+### Data Types
+
+#### `Auto`
+
+Technically, it's not a type, but just a string constant "auto" passed instead of any specific value.
+
+* "auto" can only be used where it is explicitly stated, for example, if a value can be `Movement[] | Auto`;
+* When we pass "auto" instead of something, we imply that the wallet should take care of substituting the correct value on its own:
+    * For example, `inputs: "auto"` indicates that the wallet should take care of having transaction inputs that cover its tasks;
+* Typically, the "auto" value can be found in optional fields, and this is not accidental:
+    * "auto" is the default value;
+    * If any value was omitted, the wallet should treat it as if it were equal to "auto";
```

Comment:
There is no need for introducing this value, it's better to design the constraints interface properly to cover all cases.

Reply:
I wouldn't do this if it weren't necessary and didn't form part of the concept of **this CIP**, which reflects my ideas and considerations as a developer. Moreover, almost everywhere a field that can take the value Auto can simply be omitted. Explicitly stating Auto is only needed if it's more convenient for the DApp developer.